graylog2-server/src/test/java/org/graylog2/security/ldap/LdapConnectorSSLTLSIT.java

```diff
@@ -194,11 +188,17 @@ private LdapTestConfigRequest createRequest(URI uri, boolean startTls, boolean t
         );
     }
 
-    private void assertConnectionSuccess(LdapTestConfigRequest request) throws KeyStoreException, LdapException, NoSuchAlgorithmException {
+    private void assertConnectionFailure(LdapTestConfigRequest request) {
+        Assertions.assertThatThrownBy(() -> ldapConnector.connect(request)).isNotNull();
+    }
+
+    private void assertConnectionSuccess(LdapTestConfigRequest request) throws KeyStoreException, LdapException, NoSuchAlgorithmException, IOException {
         final LdapNetworkConnection connection = ldapConnector.connect(request);
         assertThat(connection.isAuthenticated()).isTrue();
         assertThat(connection.isConnected()).isTrue();
         assertThat(connection.isSecured()).isTrue();
+
+        connection.close();
```

Comment:
How about doing this as a try-with-resource?
That ensures cleanup even if we hit an exception
```suggestion
        try (final LdapNetworkConnection connection = ldapConnector.connect(request)) {
            assertThat(connection.isAuthenticated()).isTrue();
            assertThat(connection.isConnected()).isTrue();
            assertThat(connection.isSecured()).isTrue();
        }
```